# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/Debugger-setInstrumentation-bug1571169.js
# Commit: bd40be9cdcd5
# Full Hash: bd40be9cdcd5d732a07f744b91d892470eaca264
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-08-09 03:49:20
# Description:
#   Bug 1571169 - Watch for dead object proxies for debugger objects, r=tcampbell.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41235
# ==============================================================================

diff -r 6986569d4443 -r bd40be9cdcd5 js/src/jit-test/tests/debug/Debugger-setInstrumentation-bug1571169.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/debug/Debugger-setInstrumentation-bug1571169.js	Thu Aug 08 16:09:44 2019 +0000
@@ -0,0 +1,18 @@
+// |jit-test| error: Error
+
+var lfOffThreadGlobal = newGlobal();
+evaluate(`
+  nukeAllCCWs();
+  var g92 = newGlobal({ newCompartment: true });
+  var dbg = Debugger(g92);
+  var gdbg = dbg.addDebuggee(g92);
+  gdbg.setInstrumentation(
+    gdbg.makeDebuggeeValue((kind, script, offset) => {}),
+    ["breakpoint"]
+  );
+  gdbg.setInstrumentationActive(true);
+  g92.eval(\`
+    function basic() {}
+  \`);
+`);
+